File: Python/6py 1/in_array.py
"""
|--------------------------------------------------------------------------
| Which are in?
|--------------------------------------------------------------------------
|
"""

def in_array(array1, array2):
   return sorted(set(a1 for a1 in array1 if any(a1 in a2 for a2 in array2)))


def in_array(array1, array2):
   res = []
   for a1 in array1:
      for a2 in array2:
         if a1 in a2 and not a1 in res:
            res.append(a1)
   res.sort()
   return res


def in_array(a1, a2):
   return sorted(x for x in set(a1) if any(x in y for y in a2))


print(in_array(["arp", "live", "strong"], ["lively", "alive", "harp", "sharp", "armstrong"]), ['arp', 'live', 'strong'])

File: PHP/6php 1/inArray.php
<?php
/*
|--------------------------------------------------------------------------
| Which are in?
|--------------------------------------------------------------------------
|
*/

function inArray($array1, $array2) {
   $res = [];
   foreach ($array1 as $a1) {
      foreach ($array2 as $a2) {
         if (strpos($a2, $a1) !== false) {
            $res[] = $a1;
            break;
         }
      }
   }
   sort($res);
   return $res;
}


function inArray($find, $search, $res=[]) {
   $str = " ".implode(",", $search);
   foreach($find as $x){ if( strpos($str, $x) ) array_push($res, $x); }
   sort($res); return $res;
}


function inArray($array1, $array2) {
   sort($array1);
   return array_values(array_filter($array1, function ($str1) use ($array2) {
      return strstr(implode(' ', $array2), $str1);
   }));
}


print_r(inArray(["arp", "live", "strong"], ["lively", "alive", "harp", "sharp", "armstrong"]));